src/dev/build/tasks/bundle_fleet_packages.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import JSON5 from 'json5';
+
+import { Task, read, downloadToDisk } from '../lib';
+
+const EPR_URL = 'https://epr-snapshot.elastic.co';
```

Comment:
Why is it https://epr-snapshot.elastic.co, which is considered to be unstable, rather than production endpoint https://epr.elastic.co?

Reply:
Per @simitt, we're using the snapshot registry here to support bundling "in progress" development builds of packages like APM during development/testing, and because the APM package specifically isn't usually published ahead of time. 

Maybe it'd be a better idea to pass a CLI flag here to control the EPR environment? `yarn build --use-snapshot-epr` or something along those lines to support development needs, but otherwise preferring the production environment. I'm not sure if this is suitable for APM's use case, though, so I'll let @simitt correct me if not.

Reply:
Changing default to fetch from production, but supporting a parameter to fetch from staging or snapshot SGTM. I believe it is important to be able to package non-production dev packages during the dev cycle, but I also see why people are hesitant to always fetch from a non-production stage.

Reply:
I've implemented this in https://github.com/elastic/kibana/pull/122297/commits/61b817884d1357569309973a801a66ec812b9bae and included a note for a caveat as well that I'll reiterate here.

When using the snapshot registry, all checksums provided in `fleet_package.json` will no longer be considered valid because of differences in the package stored on the snapshot registry and those in production. It's also possible that a version that exists in production does not exist in snapshot - e.g. `apm-8.0.0`. Because of this, it's necessary to remove other bundled packages when using `--use-snapshot-registry` to test bundling of development packages to prevent build failures.

I think this should be fine to support use cases like local development/testing, and we'll leave the production versions and checksums (updated in the same commit above) checked in to the Kibana repo. Just wanted to make it clear that simply opting in via that CLI flag will cause the build to fail without manual updates to `fleet_packages.json`.